File: fabric/harnesses/byte-equivalence/ts/src/serializer.ts
import * as cbor from "cbor";
import * as readline from "readline";

interface TestVector {
  version: number;
  nonce: number;
}

async function main() {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  let input = "";
  for await (const line of rl) {
    input += line;
  }

  const value: TestVector = JSON.parse(input);

  const encoded = cbor.encode(value);
  const hex = encoded.toString("hex");

  console.log(hex);
}

main().catch(console.error);

File: fabric/harnesses/byte-equivalence/rust/src/bin/serializer.rs
use std::io::{self, Read};
use mnemonic_serializer::{TestVector, serialize_to_cbor, deserialize_from_cbor};

fn main() -> io::Result<()> {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input)?;

    let value: TestVector = serde_json::from_str(&input)
        .expect("Failed to parse JSON input");

    let cbor_bytes = serialize_to_cbor(&value)
        .expect("Failed to serialize to CBOR");

    let hex = hex::encode(&cbor_bytes);
    println!("{}", hex);

    Ok(())
}
